rest-api/alembic/versions/ceb7e48928a9_add_cancelled_to_pm.py

```diff
@@ -0,0 +1,62 @@
+"""add cancelled to PM
+
+Revision ID: ceb7e48928a9
+Revises: b4f6eb55d503
+Create Date: 2018-09-25 13:07:52.661985
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import model.utils
+
+
+from participant_enums import PhysicalMeasurementsStatus, QuestionnaireStatus, OrderStatus
+from participant_enums import WithdrawalStatus, SuspensionStatus, QuestionnaireDefinitionStatus
+from participant_enums import EnrollmentStatus, Race, SampleStatus, OrganizationType, BiobankOrderStatus
+from participant_enums import MetricSetType, MetricsKey
+from model.site_enums import SiteStatus, EnrollingStatus, DigitalSchedulingStatus, ObsoleteStatus
+from model.code import CodeType
+
+# revision identifiers, used by Alembic.
+revision = 'ceb7e48928a9'
+down_revision = 'b4f6eb55d503'
+branch_labels = None
+depends_on = None
+
+
+def upgrade(engine_name):
+    globals()["upgrade_%s" % engine_name]()
+
+
+def downgrade(engine_name):
+    globals()["downgrade_%s" % engine_name]()
+
+
+
+def upgrade_rdr():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('physical_measurements', sa.Column('cancelled_status', model.utils.Enum(PhysicalMeasurementsStatus), nullable=True))
```

Comment:
Should this just be "status" with a default of COMPLETED? What will it be set to if the PM isn't cancelled? What will it be set to if the PM gets amended?

Note there's also the existing "final" field on PMs -- when you amend PMs, that gets set to false on the original row. Do we want to keep it set to true for cancelled PMs?

Reply:
When a PM is amended we set 'status' = 'amended'. I was trying to avoid any possible key errors with this. I do think final would be true on cancelled PM's. Do you think we should just use status here (it was my first choice) until I saw the _update_amended function in the DAO. I could make sure there are no conflicts in tests. What are your thoughts ?